include/bluetooth/conn.h

```diff
@@ -309,15 +310,21 @@ enum {
 	BT_CONN_ROLE_SLAVE,
 };
 
-/** Connection Info Structure */
+/** @brief Connection Info Structure
+ *
+ *  @note The connection info is available before the connection has been
+ *        established. Some of the connection information fields will
+ *        not be valid until the connection has been established and the
+ *        connected callback has been called.
```

Comment:
Without knowing what is valid, and what isn't, I don't really feel like this is helping much. 

I suggest that we either:
1) Specific exactly what is valid, and what isn't
2) Make/state everything invalid before connection is established (e.g. simply by returning `-ENOTCONN` when calling `bt_conn_get_info` for a connection that isn't established yet)

Reply:
There is still some information that you might want to use before you are connected.

It looks like it is type, role, id, and le->src that is valid before connected. Maybe easiest to document the things that are valid?

Reply:
> Maybe easiest to document the things that are valid?

It is, but require additional attention if we update the struct :) That being said, I'm okay with just adding additional documentation. 

As an addition, or alternative, we could also add "invalid value"-macros and assign those to the values that are invalid.